### ISSUE TITLE
jsx-no-target-blank warning when values are dynamic

### ISSUE THREAD

**Issue by schalkneethling** [yannickcr/eslint-plugin-react#3179]
Even though I am aware that using `target="_blank"` is not a good UX practice, we still have some areas in the codebase that, for now, still use this. My problem is not with the rule but, the fact that it is raised on the code below.

As you can see from the code, if `target="_blank"` is indeed used, we will also set `rel="noopener noreferrer"`.

```jsx
<a
  href={props.url}
  target={props.newWindow ? "_blank" : undefined}
  rel={props.newWindow ? "noopener noreferrer" : undefined}
  onClick={props.onCTAClick}
>
  {props.cta}
</a>
```

**Comment by ljharb** [comment]
The only solution here is to use an eslint override comment, or, to split it into two elements - one with target and rel, and the other with neither. The linter can’t reasonably figure out that these two separate ternaries forming 4 code branches are really 2 coupled and distinct code branches.

**Comment by schalkneethling** [comment]
> The only solution here is to use an eslint override comment, or, to split it into two elements - one with target and rel, and the other with neither. The linter can’t reasonably figure out that these two separate ternaries forming 4 code branches are really 2 coupled and distinct code branches.

Yeah, we did end up just disabling the rule here. I totally get that it would be tricky to tease this out programatically. It is interesting though to think why it is flagging this as a warning.

**Comment by ljharb** [comment]
Pretty much any rule that checks for a static prop value will warn on any dynamic value, because it can't know for sure what it will be. This is just the limits of practicality and static analysis ¯\\\_(ツ)_/¯

**Comment by schalkneethling** [comment]
@ljharb - Thank you for your feedback here. Feel free to close this is there is no path forward.
